feat: add ERC20 paymaster token cost estimation

silva-fj opened the pull request:
## Summary
- Implement token cost estimation for ERC20 paymaster operations
- Fetch real-time token prices from Binance API
- Calculate and return estimated token costs in gas estimation responses

silva-fj committed:
Make binance-api Error type public

silva-fj committed:
Add ERC20 paymaster token cost estimation

- Add TokenCostEstimate type for estimated token costs
- Implement token price fetching from Binance API
- Calculate and return token costs in gas estimation response
- Add token amount formatting with proper decimal handling

kziemianek commented on tee-worker/omni-executor/native-task-handler/src/lib.rs:
```diff
@@ -1525,16 +1640,42 @@ async fn estimate_user_op_gas(
+	};
+
+	// Step 6: Calculate token cost if ERC20 paymaster is present
+	let estimated_token_cost = if !user_op.paymasterAndData.is_empty() {
```
Does it always mean erc20 paymaster is being used ?

silva-fj replied:
No, we check if it's present in `calculate_erc20_token_cost`. Let me move the comment inside the if block to make it clear

silva-fj committed:
refactor: move comment for ERC20 paymaster logic

Relocated the comment explaining ERC20 paymaster token cost calculation
to directly precede the relevant code block for improved clarity and
code readability.

silva-fj committed:
Merge branch 'dev' into p-1703-erc20-paymaster-token-cost-estimation

Kailai-Wang commented:
The logic looks fine - I'm just wondering if this is actually required/used by backend (or if it's too verbose to calculate and return such information), I'm feeling the logic is heavier and heavier

silva-fj commented:
> The logic looks fine - I'm just wondering if this is actually required/used by backend (or if it's too verbose to calculate and return such information), I'm feeling the logic is heavier and heavier

This is something the backend could calculate on their own, they would just need to use the estimate gas rpc method then call binance api or any other api to get the exchange rate and estimate the cost in erc20 token.

silva-fj committed:
Merge branch 'dev' into p-1703-erc20-paymaster-token-cost-estimation

Combined ERC20 paymaster token cost estimation with gas fee fields from dev branch

silva-fj committed:
Fix tests after adding gas fee fields to EstimateUserOpGas

Add max_fee_per_gas and max_priority_fee_per_gas to test instances
